packages/grid/x-data-grid/src/components/base/GridOverlays.tsx

```diff
@@ -44,8 +44,6 @@ function GridOverlayWrapper(props: React.PropsWithChildren<{}>) {
         position: 'absolute',
         top: headerHeight,
         bottom: height === 'auto' ? 0 : undefined,
-        zIndex: 4, // should be above pinned columns, pinned rows and detail panel
```

Comment:
@cherniavskii shouldn't we keep `zIndex: 4` here? It was added in https://github.com/mui/mui-x/pull/4863. Without it, other overlays, e.g. no filter results, don't take the full height.

I added a background on purpose to one overlay. Here it's how it looks like:

<img width="250" src="https://user-images.githubusercontent.com/42154031/182734172-cd0824d0-7078-4ab7-aca9-d472a8e4a457.png" alt="image" style="max-width: 100%;">

Reply:
I could add it back, I thought moving it in the git patch proposed would be ok but the naming suggests that zIndex only applies to loading overlays...

How can we test this locally? Is thee a way to produce a custom NPM package? do a file:/ reference in my project's package.json...

Reply:
> How can we test this locally?

@philjones88 Please refer to the [contribution guide](https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md#how-can-i-use-a-change-that-wasnt-released-yet) for options on how to test an unreleased package.

Reply:
@m4theushw nice catch!
I think we should not keep zIndex on the overlay wrapper, because there's no way to customize it.

Instead, we can move those styles to `GridOverlay` instead:
```diff
diff --git a/packages/grid/x-data-grid/src/components/GridLoadingOverlay.tsx b/packages/grid/x-data-grid/src/components/GridLoadingOverlay.tsx
index 55ef3921f..8e7e4ea35 100644
--- a/packages/grid/x-data-grid/src/components/GridLoadingOverlay.tsx
+++ b/packages/grid/x-data-grid/src/components/GridLoadingOverlay.tsx
@@ -1,23 +1,14 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import CircularProgress from '@mui/material/CircularProgress';
-import { styled } from '@mui/material/styles';
 import { GridOverlay, GridOverlayProps } from './containers/GridOverlay';
 
-const GridLoadingOverlayRoot = styled(GridOverlay)({
-  position: 'absolute',
-  top: 0,
-  zIndex: 4, // should be above pinned columns, pinned rows and detail panel
-  width: '100%',
-  pointerEvents: 'none',
-});
-
 const GridLoadingOverlay = React.forwardRef<HTMLDivElement, GridOverlayProps>(
   function GridLoadingOverlay(props, ref) {
     return (
-      <GridLoadingOverlayRoot ref={ref} {...props}>
+      <GridOverlay ref={ref} {...props}>
         <CircularProgress />
-      </GridLoadingOverlayRoot>
+      </GridOverlay>
     );
   },
 );
diff --git a/packages/grid/x-data-grid/src/components/containers/GridOverlay.tsx b/packages/grid/x-data-grid/src/components/containers/GridOverlay.tsx
index 789704435..77146b237 100644
--- a/packages/grid/x-data-grid/src/components/containers/GridOverlay.tsx
+++ b/packages/grid/x-data-grid/src/components/containers/GridOverlay.tsx
@@ -29,8 +29,13 @@ const GridOverlayRoot = styled('div', {
   slot: 'Overlay',
   overridesResolver: (props, styles) => styles.overlay,
 })(({ theme }) => ({
-  display: 'flex',
+  position: 'absolute',
+  top: 0,
+  zIndex: 4, // should be above pinned columns, pinned rows and detail panel
+  width: '100%',
   height: '100%',
+  pointerEvents: 'none',
+  display: 'flex',
   alignSelf: 'center',
   alignItems: 'center',
   justifyContent: 'center',
```

This would make default overlays taking full height while allowing to scroll horizontally (for example to access column used for filtering that is not in the viewport).
But most importantly, it would allow to implement custom overlays with interactive elements while avoiding `pointerEvents: none` (so it won't cause another issue like #5590).

Reply:
I've tried to push the proposed changes to this branch to make sure there's no visual regressions, but turned out I don't have the permission to push to https://github.com/VQComms/mui-x/tree/remove-grid-overlays-pointer-events-style

@philjones88 could you push the changes so we trigger the CI?

Reply:
I've applied the git diff above, updated my fork and rebased this branch onto master latest.

Reply:
@philjones88 
Thanks! Can you rebase again to restart the CI? The failure in Argos CI looks like a false negative

Reply:
@cherniavskii I pushed an empty commit to trigger the CI. The branch was already up to date with master. Seems to have all passed!